internal/services/errmessages/redactor.go

```diff
@@ -0,0 +1,59 @@
+// Package errmessages formats the error messages that are sent to the client.
+package errmessages
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// RedactErrorInterceptor redacts some of the attached errors before sending it to the client.
```

Comment:
I would be more explicit that we unwrap until finding the first ErrToDisplay and returns that, or return the whole error if we didn’t find it in the chain of errors.